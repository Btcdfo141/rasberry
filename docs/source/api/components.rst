:mod:`spencerassistant.components`
===============================

air\_quality
--------------------------------------------

.. automodule:: spencerassistant.components.air_quality
    :members:
    :undoc-members:
    :show-inheritance:

alarm\_control\_panel
--------------------------------------------

.. automodule:: spencerassistant.components.alarm_control_panel
    :members:
    :undoc-members:
    :show-inheritance:

binary\_sensor
--------------------------------------------

.. automodule:: spencerassistant.components.binary_sensor
    :members:
    :undoc-members:
    :show-inheritance:

camera
---------------------------

.. automodule:: spencerassistant.components.camera
    :members:
    :undoc-members:
    :show-inheritance:

calendar
---------------------------

.. automodule:: spencerassistant.components.calendar
    :members:
    :undoc-members:
    :show-inheritance:

climate
---------------------------

.. automodule:: spencerassistant.components.climate
    :members:
    :undoc-members:
    :show-inheritance:

conversation
---------------------------

.. automodule:: spencerassistant.components.conversation
    :members:
    :undoc-members:
    :show-inheritance:

cover
---------------------------

.. automodule:: spencerassistant.components.cover
    :members:
    :undoc-members:
    :show-inheritance:

device\_tracker
---------------------------

.. automodule:: spencerassistant.components.device_tracker
    :members:
    :undoc-members:
    :show-inheritance:

fan
---------------------------

.. automodule:: spencerassistant.components.fan
    :members:
    :undoc-members:
    :show-inheritance:

light
---------------------------

.. automodule:: spencerassistant.components.light
    :members:
    :undoc-members:
    :show-inheritance:

lock
---------------------------

.. automodule:: spencerassistant.components.lock
    :members:
    :undoc-members:
    :show-inheritance:

media\_player
---------------------------

.. automodule:: spencerassistant.components.media_player
    :members:
    :undoc-members:
    :show-inheritance:

notify
---------------------------

.. automodule:: spencerassistant.components.notify
    :members:
    :undoc-members:
    :show-inheritance:

remote
---------------------------

.. automodule:: spencerassistant.components.remote
    :members:
    :undoc-members:
    :show-inheritance:

switch
---------------------------

.. automodule:: spencerassistant.components.switch
    :members:
    :undoc-members:
    :show-inheritance:

sensor
-------------------------------------

.. automodule:: spencerassistant.components.sensor
    :members:
    :undoc-members:
    :show-inheritance:

vacuum
-------------------------------------

.. automodule:: spencerassistant.components.vacuum
    :members:
    :undoc-members:
    :show-inheritance:

water\_heater
-------------------------------------

.. automodule:: spencerassistant.components.water_heater
    :members:
    :undoc-members:
    :show-inheritance:

weather
---------------------------

.. automodule:: spencerassistant.components.weather
    :members:
    :undoc-members:
    :show-inheritance:

webhook
---------------------------

.. automodule:: spencerassistant.components.webhook
    :members:
    :undoc-members:
    :show-inheritance:
