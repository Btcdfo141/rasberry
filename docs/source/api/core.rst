.. _core_module:

:mod:`spencerassistant.core`
-------------------------

.. automodule:: spencerassistant.core
    :members: