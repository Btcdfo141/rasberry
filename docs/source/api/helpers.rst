:mod:`spencerassistant.helpers`
============================

.. automodule:: spencerassistant.helpers
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.aiohttp\_client
-------------------------------------

.. automodule:: spencerassistant.helpers.aiohttp_client
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.area\_registry
------------------------------------

.. automodule:: spencerassistant.helpers.area_registry
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.check\_config
-----------------------------------

.. automodule:: spencerassistant.helpers.check_config
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.collection
--------------------------------

.. automodule:: spencerassistant.helpers.collection
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.condition
-------------------------------

.. automodule:: spencerassistant.helpers.condition
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.config\_entry\_flow
-----------------------------------------

.. automodule:: spencerassistant.helpers.config_entry_flow
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.config\_entry\_oauth2\_flow
-------------------------------------------------

.. automodule:: spencerassistant.helpers.config_entry_oauth2_flow
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.config\_validation
----------------------------------------

.. automodule:: spencerassistant.helpers.config_validation
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.data\_entry\_flow
---------------------------------------

.. automodule:: spencerassistant.helpers.data_entry_flow
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.debounce
------------------------------

.. automodule:: spencerassistant.helpers.debounce
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.deprecation
---------------------------------

.. automodule:: spencerassistant.helpers.deprecation
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.device\_registry
--------------------------------------

.. automodule:: spencerassistant.helpers.device_registry
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.discovery
-------------------------------

.. automodule:: spencerassistant.helpers.discovery
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.dispatcher
--------------------------------

.. automodule:: spencerassistant.helpers.dispatcher
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.entity
----------------------------

.. automodule:: spencerassistant.helpers.entity
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.entity\_component
---------------------------------------

.. automodule:: spencerassistant.helpers.entity_component
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.entity\_platform
--------------------------------------

.. automodule:: spencerassistant.helpers.entity_platform
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.entity\_registry
--------------------------------------

.. automodule:: spencerassistant.helpers.entity_registry
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.entity\_values
------------------------------------

.. automodule:: spencerassistant.helpers.entity_values
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.entityfilter
----------------------------------

.. automodule:: spencerassistant.helpers.entityfilter
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.event
---------------------------

.. automodule:: spencerassistant.helpers.event
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.icon
--------------------------

.. automodule:: spencerassistant.helpers.icon
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.integration\_platform
-------------------------------------------

.. automodule:: spencerassistant.helpers.integration_platform
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.intent
----------------------------

.. automodule:: spencerassistant.helpers.intent
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.json
--------------------------

.. automodule:: spencerassistant.helpers.json
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.location
------------------------------

.. automodule:: spencerassistant.helpers.location
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.logging
-----------------------------

.. automodule:: spencerassistant.helpers.logging
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.network
-----------------------------

.. automodule:: spencerassistant.helpers.network
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.restore\_state
------------------------------------

.. automodule:: spencerassistant.helpers.restore_state
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.script
----------------------------

.. automodule:: spencerassistant.helpers.script
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.service
-----------------------------

.. automodule:: spencerassistant.helpers.service
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.signal
-----------------------------

.. automodule:: spencerassistant.helpers.signal
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.state
---------------------------

.. automodule:: spencerassistant.helpers.state
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.storage
-----------------------------

.. automodule:: spencerassistant.helpers.storage
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.sun
-------------------------

.. automodule:: spencerassistant.helpers.sun
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.system\_info
----------------------------------

.. automodule:: spencerassistant.helpers.system_info
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.temperature
---------------------------------

.. automodule:: spencerassistant.helpers.temperature
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.template
------------------------------

.. automodule:: spencerassistant.helpers.template
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.translation
---------------------------------

.. automodule:: spencerassistant.helpers.translation
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.typing
----------------------------

.. automodule:: spencerassistant.helpers.typing
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.update\_coordinator
-----------------------------------------

.. automodule:: spencerassistant.helpers.update_coordinator
   :members:
   :undoc-members:
   :show-inheritance:
