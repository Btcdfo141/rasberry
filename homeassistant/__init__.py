"""Init file for Home Assistant."""


test2
