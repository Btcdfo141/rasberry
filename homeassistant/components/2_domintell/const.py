"""Constants for the Domintell Integration integration."""

DOMAIN = "2_domintell"
