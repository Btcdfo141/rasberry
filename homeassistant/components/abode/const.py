"""Constants for the Abode Security System component."""
import logging

LOGGER = logging.getLogger(__package__)

DOMAIN = "abode"
ATTRIBUTION = "Data provided by goabode.com"

DEFAULT_CACHEDB = "jaraco_abode_cache.pickle"
CONF_POLLING = "polling"
