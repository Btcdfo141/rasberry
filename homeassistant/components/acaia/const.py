"""Constants for component."""
from typing import Final

DOMAIN: Final = "acaia"
CONF_IS_NEW_STYLE_SCALE: Final = "is_new_style_scale"
