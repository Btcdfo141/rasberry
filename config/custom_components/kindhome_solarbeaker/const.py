from datetime import timedelta

from homeassistant.const import Platform

DOMAIN="kindhome_solarbeaker"
PLATFORMS = [Platform.COVER]

UPDATE_INTERVAL = timedelta(seconds=20)

SERVICE_UUID = "75c276c3-8f97-20bc-a143-b354244886d4"
TITLE="Kindhome Solarbeaker"

DATA_DEVICE = "device"
DATA_COOR = "coordinator"
