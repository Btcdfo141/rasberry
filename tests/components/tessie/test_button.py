"""Test the Tessie button platform."""
from unittest.mock import patch

from syrupy import SnapshotAssertion

from homeassistant.components.button import DOMAIN as BUTTON_DOMAIN, SERVICE_PRESS
from homeassistant.const import ATTR_ENTITY_ID, Platform
from homeassistant.core import HomeAssistant
from homeassistant.helpers import entity_registry as er

from .common import setup_platform


async def test_buttons(
    hass: HomeAssistant, snapshot: SnapshotAssertion, entity_registry: er.EntityRegistry
) -> None:
    """Tests that the button entities are correct."""

    entry = await setup_platform(hass, [Platform.BUTTON])

    entity_entries = er.async_entries_for_config_entry(entity_registry, entry.entry_id)

    assert entity_entries
    for entity_entry in entity_entries:
        assert entity_entry == snapshot(name=f"{entity_entry.entity_id}-entry")
        assert hass.states.get(entity_entry.entity_id) == snapshot(
            name=f"{entity_entry.entity_id}-state"
        )

    for entity_id, func in [
        ("button.test_wake", "wake"),
        ("button.test_flash_lights", "flash_lights"),
        ("button.test_honk_horn", "honk"),
        ("button.test_homelink", "trigger_homelink"),
        ("button.test_keyless_driving", "enable_keyless_driving"),
        ("button.test_play_fart", "boombox"),
    ]:
        with patch(
            f"homeassistant.components.tessie.button.{func}",
        ) as mock_press:
            await hass.services.async_call(
                BUTTON_DOMAIN,
                SERVICE_PRESS,
                {ATTR_ENTITY_ID: [entity_id]},
                blocking=True,
            )
            mock_press.assert_called_once()
