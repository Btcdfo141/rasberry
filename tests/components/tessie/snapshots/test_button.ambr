# serializer version: 1
# name: test_buttons[button.test_flash_lights-flash_lights][initial]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Test Flash lights',
      'icon': 'mdi:flashlight',
    }),
    'context': <ANY>,
    'entity_id': 'button.test_flash_lights',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': 'unknown',
  })
# ---
# name: test_buttons[button.test_homelink-trigger_homelink][initial]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Test Homelink',
      'icon': 'mdi:garage',
    }),
    'context': <ANY>,
    'entity_id': 'button.test_homelink',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': 'unknown',
  })
# ---
# name: test_buttons[button.test_honk_horn-honk][initial]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Test Honk horn',
      'icon': 'mdi:bullhorn',
    }),
    'context': <ANY>,
    'entity_id': 'button.test_honk_horn',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': 'unknown',
  })
# ---
# name: test_buttons[button.test_keyless_driving-enable_keyless_driving][initial]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Test Keyless driving',
      'icon': 'mdi:car-key',
    }),
    'context': <ANY>,
    'entity_id': 'button.test_keyless_driving',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': 'unknown',
  })
# ---
# name: test_buttons[button.test_play_fart-boombox][initial]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Test Play fart',
      'icon': 'mdi:volume-high',
    }),
    'context': <ANY>,
    'entity_id': 'button.test_play_fart',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': 'unknown',
  })
# ---
# name: test_buttons[button.test_wake-wake][initial]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Test Wake',
      'icon': 'mdi:sleep-off',
    }),
    'context': <ANY>,
    'entity_id': 'button.test_wake',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': 'unknown',
  })
# ---
